CarRentalService.Api.Host/Controllers/AnalyticsController.cs

```diff
@@ -0,0 +1,158 @@
+﻿using CarRentalService.Application.Contracts.Common;
+using CarRentalService.Application.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System.ComponentModel.DataAnnotations;
+
+namespace CarRentalService.Api.Host.Controllers;
+
+/// <summary>
+/// Controller for analytical data and reports
+/// Provides various analytical queries for business intelligence
+/// </summary>
+[ApiController]
+[Route("api/[controller]")]
+public class AnalyticsController(IAnalyticsService analyticsService, ILogger<AnalyticsController> logger) : ControllerBase
+{
+    /// <summary>
+    /// Gets all renters who rented vehicles of a specified model, ordered by full name
+    /// </summary>
+    /// <param name="vehicleModelId">The vehicle model ID</param>
+    /// <returns>List of renters with total spent amounts</returns>
+    [HttpGet("renters-by-model/{vehicleModelId}")]
+    [ProducesResponseType(typeof(List<RenterTotalSpentResponse>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<List<RenterTotalSpentResponse>>> GetRentersByVehicleModel(Guid vehicleModelId)
+    {
+        try
+        {
+            logger.LogInformation("Getting renters by vehicle model {VehicleModelId}", vehicleModelId);
+            var result = await analyticsService.GetRentersByVehicleModelAsync(vehicleModelId);
+            return Ok(result);
+        }
+        catch (ArgumentException ex)
+        {
+            logger.LogWarning(ex, "Invalid request for vehicle model {VehicleModelId}", vehicleModelId);
+            return BadRequest("Invalid request data");
```

Comment:
Я бы предложил в случае если это ошибка ArgumentException - возвращать сообщение данной ошибки. В нем не должно быть каких то важных деталей, а вот пользователю будет понятно в каком месте он ошибся.
```suggestion
            return BadRequest($"Invalid request data: {ex.Message}");
```